### PR TITLE
[DOCS] Documenting the support for MXFP4 data format.

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Providing info about support for MXFP4 data format in quantization on CPU. This PR addresses JIRA ticket no. 151042.

### CODE REVIEW

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/cpu-device.rst [hunk 1]
```diff
@@ -96,6 +99,7 @@ CPU plugin supports the following floating-point data types as inference precisi
 - ``f32`` (Intel® x86-64, Arm®)
 - ``bf16`` (Intel® x86-64)
 - ``f16`` (Intel® x86-64, Arm®)
+- ``MXFP4`` (Intel® x86-64)
```
Comment (yuxu42) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/cpu-device.rst:102]: actually mxfp4 is not the inference precision on CPU plugin. suggest to remove this line.

**Review by yuxu42 — CHANGES_REQUESTED** [review]

**Comment by MaximProshin** [comment]
I'm ok to have it in NNCF documentation, which is already there, but I'm not sure about OV optimization guide where we don't state anything especially preliminary support. @AlexKoff88 , please advise.

**Comment by yuxu42** [comment]
LGTM on CPU part.

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/cpu-device.rst (hunk 1 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/cpu-device.rst]: I have removed it in this commit: [4946e7154b8d677093c63f548bbf9c3bcb1fe551](https://github.com/openvinotoolkit/openvino/pull/26554/commits/4946e7154b8d677093c63f548bbf9c3bcb1fe551)

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst [hunk 2]
```diff
@@ -279,6 +279,50 @@ from Optimum Intel instead of NNCF to compress the model to INT8_ASYM.
...
++---------------+-----------------+----------------------------+
+|  Data format  |  Data type      | Description                |
++===============+=================+============================+
+| MXFP8         | | FP8 (E5M2)    | | Floating point, 8-bit    |
+|               | | FP8 (E4M3)    | | Floating point, 8-bit    |
++---------------+-----------------+----------------------------+
+| MXFP6         | | FP6 (E3M2)    | | Floating point, 6-bit    |
+|               | | FP6 (E2M3)    | | Floating point, 6-bit    |
++---------------+-----------------+----------------------------+
+| **MXFP4**     | **FP4 (E2M1)**  | **Floating point, 4-bit**  |
```
Comment (AlexKoff88) [inline docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:299]: I would let @yury-gorbachev comment if we should expose this in the guides that we have on model optimization topics. Now, MXFP4 does not have any benefits compared types (e.g. INT4, NF4) that we have.

Comment (kblaszczak-intel) [reply docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst:299]: If this should not be exposed too much, should we include it in release notes? 
We were told it is even part of marketing messaging.

So, my proposition is to move all this information to a separate article under Weight Compression. This way we have the requested description in docs but at the same time, we do not expose it anywhere beside the CPU article. We can even add some disclaimers stating that it is experimental or early adoption. What do you think?

**Review by kblaszczak-intel — APPROVED** [review]

File: docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst (hunk 2 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino-workflow/model-optimization-guide/weight-compression.rst]: I have moved the aforementioned info to a separate article, as suggested.

**Comment by kblaszczak-intel** [comment]
Merging as suggested... If there's any concern about too much exposure for the info, we can implement changes as we go...

**Pull request merged**